db/db_impl_open.cc

```diff
@@ -48,7 +48,11 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
     if (!s.ok()) {
       // No place suitable for logging
       result.info_log = nullptr;
+    } else {
+      result.info_log->SetOwner(InfoLogOwner::ROCKSDB);
     }
+  } else {
+    result.info_log->SetOwner(InfoLogOwner::CLIENT);
```

Comment:
Can we avoid to set it? SetOwner() is a virtual function that a user can override. If the user pass the same logger to multiple DBs and they are opened concurrently, we may see unexpected data race. 

By the way, even the curren implementation of PosixLogger may have a data race if multiple threads call SetOptions() because `owner_` is not atomic.

Reply:
I can avoid setting it here. I set it here because there was a place in Repairer where SanitizeOptions() was being called directly and the allocated info_log was reused, but instead I can check and set the ownership appropriately in the caller.